dotnet/src/SemanticKernel.Abstractions/Reliability/HttpRetryConfig.cs

```diff
@@ -59,7 +59,9 @@ public int MaxRetryCount
         (HttpStatusCode)HttpStatusCodeType.RequestTimeout,
         (HttpStatusCode)HttpStatusCodeType.ServiceUnavailable,
         (HttpStatusCode)HttpStatusCodeType.GatewayTimeout,
-        (HttpStatusCode)HttpStatusCodeType.TooManyRequests
+        (HttpStatusCode)HttpStatusCodeType.TooManyRequests,
+        (HttpStatusCode)HttpStatusCodeType.InternalServerError,
```

Comment:
500 is not universally retriable or not, it depends on the scenario.

Reply:
500 is typically an uncontrolled exception, returned by services that are suffering an unexpected outage. Retrying by default will hurt several services trying to recover and having at the same time to cope with extra traffic caused by clients in a retry loop.

Reply:
Correct - we should not retry on "500". 502,503, and 504 are retry-able: https://learn.microsoft.com/en-us/dotnet/architecture/cloud-native/application-resiliency-patterns

Reply:
No problem, 500 can be easily excluded from the default config and included, if needed.

Reply:
Should we consider:

425(Too Early)? 
507 InsufficientStorage?